### PR TITLE
facetimehd-firmware: 1.43_4 -> 1.43_5

### LINKED ISSUE #71952: facetimehd-firmware can't be built: url for osxupd10.11.5.dmg returns 404
```bash
$ cd ~/code/nixpkgs
$ git rebase -i channels/nixos-unstable # be on nixos-unstable
$ nix-build -A facetimehd-firmware

these derivations will be built:
  /nix/store/93p5ac1jahin53k6ww7mi4af74qv4324-unpack.drv
  /nix/store/bpm9g699manqjnj8l04grd715dhzk7i1-osxupd10.11.5.dmg.drv
  /nix/store/mx6cr7r0zw08d35jiab72han6whjlgwn-facetimehd-firmware-1.43_4.drv
building '/nix/store/93p5ac1jahin53k6ww7mi4af74qv4324-unpack.drv'...
building '/nix/store/bpm9g699manqjnj8l04grd715dhzk7i1-osxupd10.11.5.dmg.drv'...

trying https://support.apple.com/downloads/DL1877/en_US/osxupd10.11.5.dmg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404 Not Found
error: cannot download osxupd10.11.5.dmg from any mirror
builder for '/nix/store/bpm9g699manqjnj8l04grd715dhzk7i1-osxupd10.11.5.dmg.drv' failed with exit code 1
cannot build derivation '/nix/store/mx6cr7r0zw08d35jiab72han6whjlgwn-facetimehd-firmware-1.43_4.drv': 1 dependencies couldn't be built
error: build of '/nix/store/mx6cr7r0zw08d35jiab72han6whjlgwn-facetimehd-firmware-1.43_4.drv' failed


```

### PR DESCRIPTION
**Pull request by offlinehacker**
###### Motivation for this change

Fixes: #71952
Fix NixOS kernel as it does not build due missing firmware link. I can't be sure if offsets of firmware inside file are correct  (i'm sure downloaded ranges are correct) as i cannot test as i do not have required hardware, but i guess it should do the trick.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @womfoo @grahamc

### CODE REVIEW

**Comment by offlinehacker** [comment]
See also: https://github.com/patjak/facetimehd-firmware/issues/1

**Review by veprbl — APPROVED** [review]
Passes nix-review (x86_64-linux, NixOS)
```
1 package were build:
facetimehd-firmware
```

**Comment by offlinehacker** [comment]
Should be fine since it produces firmware with same checksum, didn't noticed it before: https://github.com/patjak/facetimehd-firmware/pull/2/files#diff-634903fbf2f39c35200155b6e130caacR22

**Review by womfoo — APPROVED** [review]
Tested OK on my MacBookPro11,3 running nixos-unstable. This also needs to be backported to 19.09.

Thanks for looking into this @offlinehacker!

File: pkgs/os-specific/linux/firmware/facetimehd-firmware/default.nix [hunk 1]
```diff
@@ -2,16 +2,16 @@
...
 
   # Updated according to https://github.com/patjak/bcwc_pcie/pull/81/files
   # and https://github.com/patjak/bcwc_pcie/blob/5a7083bd98b38ef3bd223f7ee531d58f4fb0fe7c/firmware/Makefile#L3-L9
   # and https://github.com/patjak/bcwc_pcie/blob/5a7083bd98b38ef3bd223f7ee531d58f4fb0fe7c/firmware/extract-firmware.sh
 
   # From the Makefile:
-  dmgUrl = "https://support.apple.com/downloads/DL1877/en_US/osxupd10.11.5.dmg";
-  dmgRange = "205261917-208085450"; # the whole download is 1.3GB, this cuts it down to 2MB
+  dmgUrl = "https://updates.cdn-apple.com/2019/cert/041-88431-20191011-e7ee7d98-2878-4cd9-bc0a-d98b3a1e24b1/OSXUpd10.11.5.dmg";
+  dmgRange = "204909802-207733123"; # the whole download is 1.3GB, this cuts it down to 2MB
```
Comment (thpham) [inline pkgs/os-specific/linux/firmware/facetimehd-firmware/default.nix]: How do you define this range ?

**Comment by thpham** [comment]
LGTM, I tested on My MBP 15 (2015) and it works.

File: pkgs/os-specific/linux/firmware/facetimehd-firmware/default.nix (hunk 1 above)
Comment (veprbl) [reply pkgs/os-specific/linux/firmware/facetimehd-firmware/default.nix]: @thpham https://github.com/patjak/facetimehd-firmware/issues/1#issuecomment-549585472

Comment (offlinehacker) [reply pkgs/os-specific/linux/firmware/facetimehd-firmware/default.nix]: See here: https://github.com/patjak/facetimehd-firmware/issues/1#issuecomment-549585472

Comment (offlinehacker) [reply pkgs/os-specific/linux/firmware/facetimehd-firmware/default.nix]: Basically the idea is that you find xz streams and offsets of these inside dmg file and then get files inside these streams and find a matching one. For end range I did bissection based on if I get required file from stream or not

**Comment by alexeymuranov** [comment]
Still needs to be ported to stable, i suppose...

**Pull request merged**